From b3a59fb97707901294245d799592beda70a126f7 Mon Sep 17 00:00:00 2001
From: Vinith Kumar R <quic_vinithku@quicinc.com>
Date: Tue, 1 Oct 2024 12:20:16 +0530
Subject: [PATCH 3/4] wifi: ath12k: Add NULL check for sta to avoid explicit
 dereference crash in ieee80211_xmit_fast_finish

Currently, in the ieee80211_xmit_fast_finish function call, sta
argument is dereferred explicitly. This will cause a crash if sta
is NULL. So avoid the possible exception case by adding a NULL check
before the ieee80211_xmit_fast_finish function call.

Signed-off-by: Vinith Kumar R <quic_vinithku@quicinc.com>
---
 net/mac80211/tx.c | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/net/mac80211/tx.c b/net/mac80211/tx.c
index 0d0d527..879e78a 100644
--- a/net/mac80211/tx.c
+++ b/net/mac80211/tx.c
@@ -4001,9 +4001,10 @@ begin:
 		    (tx.key->conf.flags & IEEE80211_KEY_FLAG_GENERATE_IV))
 			pn_offs = ieee80211_hdrlen(hdr->frame_control);
 
-		r = ieee80211_xmit_fast_finish(sta->sdata, sta, pn_offs,
-					       tx.key, &tx);
-		if (r != TX_CONTINUE) {
+		if(sta)
+			r = ieee80211_xmit_fast_finish(sta->sdata, sta, pn_offs,
+						       tx.key, &tx);
+		if (r != TX_CONTINUE || !sta) {
 			ieee80211_free_txskb(&local->hw, skb);
 			goto begin;
 		}
-- 
2.34.1

